napari/layers/labels/_tests/test_labels_mouse_bindings.py

```diff
@@ -0,0 +1,38 @@
+import numpy as np
+import collections
+from napari.layers import Labels
+from napari.utils.interactions import (
+    ReadOnlyWrapper,
+    mouse_press_callbacks,
+    mouse_move_callbacks,
+    mouse_release_callbacks,
+)
+
+
+def test_mouse_move():
+    """Test painting labels with different brush sizes."""
+    np.random.seed(0)
+    data = np.random.randint(20, size=(20, 20))
+    layer = Labels(data)
+    layer.brush_size = 10
+    layer.mode = 'paint'
+    layer.selected_label = 3
+    layer._last_cursor_coord = (0, 0)
+    layer.coordinates = (19, 19)
+
+    Event = collections.namedtuple('Event', 'type is_dragging')
+
+    # Simulate click
+    event = ReadOnlyWrapper(Event(type='mouse_press', is_dragging=False))
+    mouse_press_callbacks(layer, event)
+
+    # Simulate drag
+    event = ReadOnlyWrapper(Event(type='mouse_move', is_dragging=True))
+    mouse_move_callbacks(layer, event)
+
+    # Simulate release
+    event = ReadOnlyWrapper(Event(type='mouse_release', is_dragging=False))
+    mouse_release_callbacks(layer, event)
+
+    assert np.unique(layer.data[:5, :5]) == 3
```

Comment:
@sofroniewn , can you please explain why `np.unique(layer.data[:5, :5]) == 3`?

This is my understanding from [the paint callback](https://github.com/napari/napari/blob/6313ffc2dcc2408fb56eb5b132399af34c126708/napari/layers/labels/_labels_mouse_bindings.py#L21)

1. (line 26) On the first click, label 3 is painted on `layer.coordinates` (with brush size 10) and `layer._last_cursor_coord` is set to `layer.coordinates` (19, 19)
2. (line 30) The dragging event starts. `layer.coordinates` isn't updated, so no new additional pixels are painted
3. (line 34) The dragging and clicking ends.

If that is correct, it seems like `np.unique(layer.data[-5:, -5:]) == 3` should be true, but I can't tell how the pixels at `[:5, :5]` are painted.

Reply:
`layer._last_cursor_coord = (0, 0)` so the painting goes from `(0, 0)` to `(19, 19)`. I will add a note explaining

Reply:
Sorry to dwell on this - I am mostly curious for my own understanding.

I see that `layer._last_cursor_coord` is set to `(0, 0)` on [line 19](https://github.com/napari/napari/blob/1d5210e3b50a2746f432695976a02fa59b4e7250/napari/layers/labels/_tests/test_labels_mouse_bindings.py#L19). However, in the `paint()` callback, it looks like when the mouse is clicked, [the pixels within the brush radius of `layer.coordinates` (`(19, 19)`) are painted](https://github.com/napari/napari/blob/1d5210e3b50a2746f432695976a02fa59b4e7250/napari/layers/labels/_labels_mouse_bindings.py#L9) and then [`last_cursor_coord` (now local to `paint()`) is assigned `layer.coordinates`](https://github.com/napari/napari/blob/1d5210e3b50a2746f432695976a02fa59b4e7250/napari/layers/labels/_labels_mouse_bindings.py#L10) (i.e., `last_cursor_coord == (19, 19)`) before yielding. Therefore, to me, it seems like the dragging should start at (`(19, 19)`).

Reply:
Now that we dropped `_last_cursor_coord` it's a little weird that this worked, it was because not all the right updates were being called. I've fixed the test to have two explicit calls to `layer.position` (which is something like `worldslice coordinates` in our new (but yet to be implemented) terminology) which triggers the appropriate coordinate and value updates behind the scene. When I do the world coordinates PR we'll sort out the names of things like position and coordinates which is super confusing right now

Reply:
Thanks, @sofroniewn - that is helpful! I'm looking forward to world coordinates!